### PR TITLE
E2E test support for Z clusters

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

File: tests/e2e/iks-cluster [hunk 1]
```diff
@@ -68,7 +68,7 @@ PLATFORM_kubernetes="k8s"
 PLATFORM_openshift="ocp"
 
 PRIMARY_ZONE_ID="01"	# ZONE_vpc_gen2_01 / ZONE_classic_01
-SECOND_ZONE_ID="03"	# ZONE_vpc_gen2_03 / ZONE_classic_03
+SECOND_ZONE_ID="02"	# ZONE_vpc_gen2_03 / ZONE_classic_03
```
Comment (sameshai) [inline tests/e2e/iks-cluster:71]: @GunaKKIBM change comment accordingly

Comment (ambiknai) [reply tests/e2e/iks-cluster:71]: @GunaKKIBM  Could you explain reason for this change.

**Review by ambiknai — COMMENTED** [review]
Signed commit required. Could you please check comments. 

Thanks

File: tests/e2e/iks-cluster [hunk 2]
```diff
@@ -214,8 +214,9 @@ function check_addon_pod_status {
   pod_name=$1
   while true; do
     attempts=$((attempts+1))
-    pod_status=$(kubectl get sts -n kube-system | grep "$pod_name"| awk '{print $2}')
-    if [   "$pod_status" = "1/1" ]; then
+    pod_status_sts=$(kubectl get sts -n kube-system | grep "$pod_name"| awk '{print $2}')
+    pod_status_deployment=$(kubectl get deployment -n kube-system | grep "$pod_name"| awk '{print $2}')
```
Comment (GunaKKIBM) [inline tests/e2e/iks-cluster:218]: Grep only for controller pods and node server pods directly and check status

Comment (GunaKKIBM) [reply tests/e2e/iks-cluster:218]: https://github.ibm.com/alchemy-containers/armada-storage/issues/4875 - created GHE to track this

**Review by prankulmahajan — APPROVED** [review]
lgtm

**Review by sameshai — APPROVED** [review]

**Pull request merged**